packages/data/src/navigation/actions.js

```diff
@@ -23,3 +29,118 @@ export function addMenuItems( menuItems ) {
 		menuItems,
 	};
 }
+
+export function getFavoritesFailure( error ) {
+	return {
+		type: TYPES.GET_FAVORITES_FAILURE,
+		error,
+	};
+}
+
+export function getFavoritesRequest( favorites ) {
+	return {
+		type: TYPES.GET_FAVORITES_REQUEST,
+		favorites,
+	};
+}
+
+export function getFavoritesSuccess( favorites ) {
+	return {
+		type: TYPES.GET_FAVORITES_SUCCESS,
+		favorites,
+	};
+}
+
+export function addFavoriteRequest( favorite ) {
+	return {
+		type: TYPES.ADD_FAVORITE_REQUEST,
+		favorite,
+	};
+}
+
+export function addFavoriteFailure( favorite, error ) {
+	return {
+		type: TYPES.ADD_FAVORITE_FAILURE,
+		favorite,
+		error,
+	};
+}
+
+export function addFavoriteSuccess( favorite ) {
+	return {
+		type: TYPES.ADD_FAVORITE_SUCCESS,
+		favorite,
+	};
+}
+
+export function removeFavoriteRequest( favorite ) {
+	return {
+		type: TYPES.REMOVE_FAVORITE_REQUEST,
+		favorite,
+	};
+}
+
+export function removeFavoriteFailure( favorite, error ) {
+	return {
+		type: TYPES.REMOVE_FAVORITE_FAILURE,
+		favorite,
+		error,
+	};
+}
+
+export function removeFavoriteSuccess( favorite, error ) {
+	return {
+		type: TYPES.REMOVE_FAVORITE_SUCCESS,
+		favorite,
+		error,
+	};
+}
+
+export function* addFavorite( favorite ) {
+	yield addFavoriteRequest( favorite );
+	yield addFavoriteSuccess( favorite );
```

Comment:
I'm not sure why we're calling `addFavoriteSuccess()` before the request here?

Reply:
Good catch, must have been left over from testing.  Removed in https://github.com/woocommerce/woocommerce-admin/pull/6275/commits/0f15a74afe5a2369c244c53bea6815e13cf1fb7b